### PR TITLE
Fold System.Private.Reflection.Core into CoreLib

### PR DESCRIPTION
**Pull request by jkotas**

### CODE REVIEW

File: src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets [hunk 1]
```diff
@@ -109,7 +109,6 @@ The .NET Foundation licenses this file to you under the MIT license.
   </ItemGroup>
 
   <ItemGroup>
-    <_ExcludedPrivateSdkAssemblies Include="$(IlcSdkPath)System.Private.Reflection.Core.dll" Condition="$(IlcDisableReflection) == 'true'" />
     <PrivateSdkAssemblies Include="$(IlcSdkPath)*.dll" Exclude="@(_ExcludedPrivateSdkAssemblies)"/>
```
Comment (MichalStrehovsky) [inline src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets:112]: Let's delete `_ExcludedPrivateSdkAssemblies` too

**Review by MichalStrehovsky — APPROVED** [review]
Thanks!

**Comment by MichalStrehovsky** [comment]
The warning that is failing the build is https://github.com/dotnet/linker/issues/2790.

It complains that we're creating a delegate to a method with annotations. Trimming cannot enforce the delegate will be always called from places that are analyzable.

A good fix might be to just remove the annotations on the method and add a warning suppression instead (it's going to warn that we call GetUninitializedObject). It's always safe to call GetUninitializedObject on an underlying type of a nullable. We also have a test for that being added in https://github.com/dotnet/runtime/pull/70201 (`TestVTableOfNullableUnderlyingTypes`).

**Comment by jkotas** [comment]
Failure is #70626

**Pull request merged**